client/src/components/course-view/Assignments.js

```diff
@@ -11,15 +11,15 @@ import Typography from '@mui/material/Typography';
 import AssignmentCategory from '../assignments/AssignmentCategory';
 
 
-const Assignments = ({ assignments, formula, instance }) => {
+const Assignments = ({ assignments, formula, instance, courseCode }) => {
   const navigate = useNavigate();
   
   return (
     <Box borderRadius={1} sx={{ bgcolor: 'primary.light', p: 1.5, display: 'inline-block' }}>
       <Typography variant='h6' align='left' sx={{ ml: 1.5 }} >Study Attainments</Typography>
       <Box sx={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', alignItems: 'center', pb: 1 }}>
         <Typography align='left' sx={{ ml: 1.5 }} >{'Grading Formula: ' + formula}</Typography>
-        <Button onClick={ () => navigate('/select-formula') }>Edit formula</Button>
+        <Button onClick={() => navigate(`/${courseCode}/select-formula/${instance.id}`) }>Edit formula</Button>
```

Comment:
A comment could be added here that one the `courseId` can be fetched from the path instead of `courseCode`, then this should be changed.